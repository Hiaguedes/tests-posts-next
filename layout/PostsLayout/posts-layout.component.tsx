import React from 'react';
import { LayoutContainer, LayoutHeader, Logo } from './posts-layout.styles';
import Link from 'next/link';

interface LayoutProps {
    children: React.ReactNode;
}

const PostsLayout = ({children}: LayoutProps) => {
    return (
        <LayoutContainer>
            <LayoutHeader>
            <Logo>
                <Link href="/">
                    <a style={{width:'100%', height:'100%', display: 'flex', justifyContent: 'center', alignItems: 'center'}}>
                        Logo
                    </a>
                </Link>
            </Logo>
            </LayoutHeader>
                {children}
        </LayoutContainer>
    );
}

export default PostsLayout;
