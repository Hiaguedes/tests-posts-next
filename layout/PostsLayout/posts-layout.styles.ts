import styled from 'styled-components';

export const LayoutContainer = styled.div`
    width: 100vw;
    height: 100vh;
    overflow-y: auto;
`;

export const LayoutHeader = styled.div`
    margin: 2rem 5rem;
    height: 15%;
    display: flex;
    align-items: center;
    justify-content: space-between;
`;

export const Logo = styled.div`
    height: 80%;
    border: 1px solid black;
    width: 10rem;
    display: flex;
    justify-content: center;
    align-items: center;
`;