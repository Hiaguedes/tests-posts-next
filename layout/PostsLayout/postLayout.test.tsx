import { screen, render, waitFor } from '@testing-library/react';
import PostsLayout from './posts-layout.component';
import '@testing-library/jest-dom'
describe(('No layout da página de posts'), () => {

    test('A logo aparece', () => {
        render(<PostsLayout>a</PostsLayout>)
        expect(screen.getByTestId('logo')).toBeInTheDocument();
    })

    test('Snapshot do componente é o mesmo 20/01/2021', () => {
        const {container} = render(<PostsLayout>a</PostsLayout>)
        expect(container.firstChild).toMatchSnapshot();
    })
})