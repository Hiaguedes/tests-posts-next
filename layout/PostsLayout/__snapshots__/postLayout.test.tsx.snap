// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No layout da página de posts Snapshot do componente é o mesmo 20/01/2021 1`] = `
<div
  class="posts-layoutstyles__LayoutContainer-tuujpg-0 hVbdt"
>
  <div
    class="posts-layoutstyles__LayoutHeader-tuujpg-1 fLHURT"
  >
    <div
      class="posts-layoutstyles__Logo-tuujpg-2 FPlKk"
      data-testid="logo"
    >
      <a
        href="/"
        style="width: 100%; height: 100%; display: flex; justify-content: center; align-items: center;"
      >
        Logo
      </a>
    </div>
  </div>
  a
</div>
`;
